### ISSUE TITLE
Nginx Reverse proxying / Multiple HTTP servers

### ISSUE THREAD

**Issue by Tom-Bonnike** [indutny/sticky-session#47]
Hi.
My issue is summed up here: http://stackoverflow.com/questions/37911155/nodejs-clusters-socket-io-redis-nginx
I am aware of issue #24 , issue #6  and PR #45 , but they sadly haven't been answered.

What I want to know is if what I'm trying to do is even possible? If it isn't, what should I do instead? 
Wouldn't just allowing the websockets transport be a good solution to avoid the multiple requests?

**Comment by v-sevostyanov** [comment]
I think you should use https://github.com/indutny/sticky-session with https://github.com/socketio/socket.io-redis

Something like this:

```
const cluster = require('cluster');
const sticky = require('sticky-session');
const io_redis = require('socket.io-redis');

const port = process.env.PORT || 3000;
const http = require('http');
const app = require('../app');
const server = http.createServer(app);
const io = require('socket.io')(server);

io.adapter(io_redis({ host: 'localhost', port: 6379 }));

if (!sticky.listen(server, port)) {
    server.once('listening', function() {
        console.log('server started on port: ' + port);
    });

    cluster.on('online', (worker) => {
        console.log('worker ' + worker.id + ' pid ' + process.pid + ' started');
    });

    cluster.on('disconnect', (worker) => {
        console.log(`The worker #${worker.id} has disconnected`);
    });

    cluster.on('exit', (worker, code, signal) => {
        console.log('worker %d died (%s). restarting...', worker.process.pid, signal || code);
    });
} else {
    require('../sockets/server')(io);
}
```

Of course, you should install redis server. http://redis.io/download

**Comment by v4l3r10** [comment]
I can confirm the suggestion of @Mad-Head  . I use the same approch in a production env  and works very well!

**Comment by Tom-Bonnike** [comment]
Yes, but the problem is that sticky session doesn't work with proxies (as it uses IP to transfer the requests to the right cluster, it would always show 127.0.0.1 as the IP). I could be mistaken though, and I haven't even tried because I actually just solved my problem.

I decided to drop the longpolling transport for socket.io and now only use websocket transport, which only needs 1 request to make the initial connection. This means it won't work on some platforms, but I don't really care for my project. So I don't need to use sticky-sessions anymore, or any other code that would help me retrieve and store the real IP of the original request.

Thanks anyway to you two!
If you feel like this is wrong, feel free to explain why! :)
